src/Package/Impl/Sql/Publish/SqlPublshOptionsDialog.xaml.cs

```diff
@@ -18,27 +19,53 @@ namespace Microsoft.VisualStudio.R.Package.Sql.Publish {
     /// Interaction logic for SqlPublsh.xaml
     /// </summary>
     public partial class SqlPublshOptionsDialog : DialogWindow {
-        private readonly SqlPublishOptionsDialogViewModel _model;
         private readonly IApplicationShell _appShell;
         private readonly IProjectSystemServices _pss;
+        private readonly IProjectConfigurationSettingsProvider _pcsp;
+        private SqlPublishOptionsDialogViewModel _model;
 
-        public SqlPublshOptionsDialog(IApplicationShell appShell, IProjectSystemServices pss, IProjectConfigurationSettingsProvider pcsp) :
+        public static async Task<SqlPublshOptionsDialog> CreateAsync(
+            IApplicationShell appShell, IProjectSystemServices pss, 
+            IProjectConfigurationSettingsProvider pcsp) {
+            var dialog = new SqlPublshOptionsDialog(appShell, pss, pcsp);
+            await dialog.InitializeAsync();
+            return dialog;
+        }
+
+        public static async Task<SqlPublshOptionsDialog> CreateAsync(
+            IApplicationShell appShell, IProjectSystemServices pss,
+            IFileSystem fs, IProjectConfigurationSettingsProvider pcsp) {
+            var dialog = new SqlPublshOptionsDialog(appShell, pss, fs, pcsp);
+            await dialog.InitializeAsync();
+            return dialog;
+        }
+
+        private SqlPublshOptionsDialog(IApplicationShell appShell, IProjectSystemServices pss, IProjectConfigurationSettingsProvider pcsp) :
             this(appShell, pss, new FileSystem(), pcsp) {
             InitializeComponent();
+
+            TargetTypeList.SelectedIndex = _model.SelectedTargetTypeIndex;
+            TargetList.SelectedIndex = _model.SelectedTargetIndex;
+            CodePlacementList.SelectedIndex = _model.SelectedQuoteTypeIndex;
```

Comment:
I removed all bindings to the index since selection change notification is all we need